src/app/components/import/ImportModal.js

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import { useNotifications, FormModal, useUser, useUserKeys } from 'react-components';
+
+import ImportFooter from './ImportFooter';
+import AttachingModalContent from './AttachingModalContent';
+import ImportCsvModalContent from './ImportCsvModalContent';
+import ImportingModalContent from './ImportingModalContent';
+import ImportGroupsModalContent from './ImportGroupsModalContent';
+
+import { IMPORT_STEPS } from '../../constants';
+import { hasCategories } from '../../helpers/import';
+import { readCsv } from '../../helpers/csv';
+import { readFileAsString } from 'proton-shared/lib/helpers/file';
+
+const { ATTACHING, ATTACHED, CHECKING_CSV, IMPORTING, IMPORT_GROUPS } = IMPORT_STEPS;
+
+const getI18nTitle = {
+    [ATTACHING]: c('Title').t`Import contacts`,
+    [ATTACHED]: c('Title').t`Import contacts`,
+    [CHECKING_CSV]: c('Title').t`Import CSV file`,
+    [IMPORTING]: c('Title').t`Importing contacts`,
+    [IMPORT_GROUPS]: c(`Title`).t`Import groups`
+};
+
+const ImportModal = ({ onClose, ...rest }) => {
+    const { createNotification } = useNotifications();
+    const [user] = useUser();
+    const [userKeysList, loadingUserKeys] = useUserKeys(user);
+
+    const [step, setStep] = useState(ATTACHING);
+    const [file, setFile] = useState({ attached: null, read: null });
+    const [vcardContacts, setVcardContacts] = useState([]);
+    const [encryptingDone, setEncryptingDone] = useState(false);
+
+    const handleAttach = async ({ target }) => {
+        // TODO: set some limit on the the size ?
+        const attachedFile = [...target.files].filter(({ type }) => ['text/vcard', 'text/csv'].includes(type))[0];
+
+        if (target.files.length && !attachedFile) {
+            return createNotification({
+                type: 'error',
+                text: c('Error notification').t`No .csv or .vcard file selected`
+            });
+        }
+        setStep(ATTACHED);
+        setFile({ ...file, attached: attachedFile });
+    };
+
+    const handleClear = () => {
+        setFile({ attached: null, read: null });
+        setStep(ATTACHING);
+    };
+
+    const handleEncryptingDone = () => setEncryptingDone(true);
+
+    const handleSubmit = {
```

Comment:
Rather than having separate control flows for submit, section, and footer, I prefer doing this because it keeps the behavior closer together:

```
    const { section, ...modalProps } = (() => {
        if (step <= STEPS. ATTACHING) {
            const handleSubmit = async () => {
              ........
            };

            return {
                section: <AttachingModalContent
                    attached={step === ATTACHED}
                    file={file.attached}
                    onAttach={handleAttach}
                    onClear={handleClear}
                />,
                title: c('Action').t`My title`
                submit: c('Action').t`Next`
                onSubmit: () => withLoading(handleSubmit())
            };
        }
       ...
       )();

       return (<FormModal
            {...modalProps}
            {...rest}
        >
          {section}
         ...)
```